### PR TITLE
Full split function

### PR DESCRIPTION
**Pull request by Ly0n**
A complete revision of the split and gather function to be better readable:

* Allows to the configure full split or random split for the main project (solves #44)
* Random selection of dependency contributors (solves #62)
* Removed most content of doc folder since it it is not used

### CODE REVIEW

File: openselery/configuration.py [hunk 1]
```diff
@@ -25,6 +26,9 @@ class OpenSeleryConfig(object):
         "number_payout_contributors_per_run": 1,
         "max_payout_per_run": 0.000002,
 
+        "random_split": False,
+        "full_split": False,
+
```
Comment (cornerman) [inline openselery/configuration.py:31]: Maybe better to introduce one option for all random modes:

like `random_mode: "split"` or `random_mode: "full"`. Now with boolean some combinations make no sense like `true/true`.

File: selery.yml [hunk 2]
```diff
@@ -6,11 +6,13 @@ simulation: False
...
 ## include the dependencies of the target folder
-include_dependencies: False 
+include_dependencies: True 
 ## include the tooling and runtime from file
-include_tooling_and_runtime: False 
+include_tooling_and_runtime: False
 ## minimum contributions made to be considered
 min_contributions: 1
+## number of randomly selected dependency contributors
+included_dependency_contributor: 2
```
Comment (cornerman) [inline selery.yml:15]: What would be a good sane default? What is the default if the value is not set?

File: selery.yml [hunk 3]
```diff
@@ -21,18 +23,22 @@ consider_releases: True
...
 ## release weights
 release_weight: 70
+## weighted split over all contributors
+full_split: False
+## random weighted split with equal payout per contributor
+random_split: True
+## the amount of btc send per user at random_split
+btc_per_transaction: 0.000002
+## numbers of contributors payout per run at random split
+number_payout_contributors_per_run: 5
```
Comment (cornerman) [inline selery.yml:33]: This count includes dependency contributors right?

Comment (Ly0n) [reply selery.yml:33]: Yes, maybe the comment should me more clear or the name better

File: openselery/configuration.py (hunk 1 above)
Comment (kikass13) [reply openselery/configuration.py:31]: true, 

there shoudl be a list of possible contribution "modes" to 

- give things names
-  remove complexity
- create transparency regarding the "possibilities" of payout

There should be no manual mode for choosing people regarding non defined metrics. And if there is such desire, it should be made obvious and clear via

 ``` random_mode: "custom" ``` or something like that

File: openselery/openselery.py [hunk 4]
```diff
@@ -175,99 +193,141 @@ def gather(self):
...
 
                             # safe project / dependency information
-                            generalProjects.append(gitproject)
-                            generalDependencies.extend(libIoDependencies)
+                            dependencyProjects.append(gitproject)
+
+            for p in dependencyProjects:
+                # grab contributors
+                depContributors = self.githubConnector.grabRemoteProjectContributors(
+                p)
```
Comment (kikass13) [inline openselery/openselery.py:201]: this newline style of your IDE (or your own weird thing) annoys the hell out of me :D

i mean .. really? 
```
depContributors = self.githubConnector.grabRemoteProjectContributors(
                p)
```

xD

File: openselery/openselery.py [hunk 5]
```diff
@@ -175,99 +193,141 @@ def gather(self):
...
+                # filter contributors
+                depContributors = selery_utils.validateContributors(
+                    depContributors, self.config.min_contributions)
+            # safe contributor information
```
Comment (kikass13) [inline openselery/openselery.py:205]: wrong indentation comment

File: openselery/openselery.py [hunk 6]
```diff
@@ -175,99 +193,141 @@ def gather(self):
...
             raise Exception("Aborting")
 
-        recipients = random.choices(
-            contributors, weights, k=self.config.number_payout_contributors_per_run)
-        for contributor in recipients:
-            self.log(" -- '%s': '%s' [w: %s]" % (contributor.stats.author.html_url,
-                                              contributor.stats.author.name, weights[contributors.index(contributor)]))
-            self.log("  > via project '%s'" % contributor.fromProject)
-        return recipients
+        if self.config.random_split and not self.config.full_split:
```
Comment (kikass13) [inline openselery/openselery.py:301]: @cornerman already said it but ill repeat it here:

the splitting / choosing of contributors should be based on a single defined keyword regarding a single obvious rule:
`self.config.split_mode` therefore should be a string containing the "action" to take, so `random`, `full`, `custom`, `main_only`, and so on ...

File: openselery/configuration.py (hunk 1 above)
Comment (Ly0n) [reply openselery/configuration.py:31]: Very good point. I will implement this suggestion.

**Pull request merged**